kuka_rsi_hw_interface/src/kuka_hardware_interface_node.cpp

```diff
@@ -54,62 +54,43 @@ int main(int argc, char** argv)
   kuka_rsi_hw_interface.configure();
 
   // Set up timers
-  struct timespec ts = {0, 0};
-  if (clock_gettime(CLOCK_MONOTONIC, &ts) != 0)
-  {
-    ROS_FATAL("Failed to poll clock!");
-  }
-  ros::Time last(ts.tv_sec, ts.tv_nsec);
-  ros::Time now(ts.tv_sec, ts.tv_nsec);
-  ros::Duration period(1.0);
+  ros::Time ros_now;
+  ros::Duration ros_period;
+  auto rt_last = std::chrono::steady_clock::now();
+  auto rt_now = rt_last;
 
   controller_manager::ControllerManager controller_manager(&kuka_rsi_hw_interface, nh);
 
   kuka_rsi_hw_interface.start();
 
   // Get current time and elapsed time since last read
-  if (!clock_gettime(CLOCK_MONOTONIC, &ts))
-  {
-    now.sec = ts.tv_sec;
-    now.nsec = ts.tv_nsec;
-    period = now - last;
-    last = now;
-  }
-  else
-  {
-    ROS_FATAL("Failed to poll clock!");
-  }
+  ros_now = ros::Time::now();
+  rt_now = std::chrono::steady_clock::now();
+  ros_period.fromSec(std::chrono::duration_cast<std::chrono::duration<double>>(rt_now - rt_last).count());
```

Comment:
Well, I think I'd prefer to let the variable name reflect it's function / origin, as the type can be found by looking up it's declaration.

Your PR introduces the concept of using two separate timebases, so any durations would also be in any of those two. If we then name such a duration in a way that refers to the timebase it doesn't actually originate from, it seems strange.